source/common/formatter/http_specific_formatter.cc

```diff
@@ -197,6 +197,17 @@ HeadersByteSizeFormatter::formatValueWithContext(const HttpFormatterContext& con
       context.requestHeaders(), context.responseHeaders(), context.responseTrailers()));
 }
 
+ProtobufWkt::Value TraceIDFormatter::formatValueWithContext(const HttpFormatterContext& context,
+                                                            const StreamInfo::StreamInfo&) const {
+  return ValueUtil::stringValue(context.activeSpan().getTraceIdAsHex());
+}
+
+absl::optional<std::string>
+TraceIDFormatter::formatWithContext(const HttpFormatterContext& context,
+                                    const StreamInfo::StreamInfo&) const {
+  return context.activeSpan().getTraceIdAsHex();
+}
+
```

Comment:
I think it would be more consistent with the existing commands if a missing trace id returned `SubstitutionFormatUtils::unspecifiedValue()` and `absl::nullopt` from these methods.

Reply:
update